EL-3009

PearseMcMurray opened the pull request:
New Angular Notifications documentation section.


https://jira.autonomy.com/browse/EL-3009

ashley-hunter commented on docs/app/pages/components/components-sections/notifications/notifications/notifications.component.html:
```diff
@@ -0,0 +1,61 @@
+<p>The <code>notificationService</code> has several functions available to control notifications.</p>
+
+<uxd-api-properties>
+    <tr uxd-api-property name="showNotification">
```
I know this is assigned to Donna, but I just had a quick read over and have a few minor suggestions.

Consider removing 'notification' from the function names as they are going to be calling `notificationService.xyz()` the notificationService already implies the context is regarding notifications so it can be removed from the function names. Eg:

showNotification -> show
dismissNotification -> dismiss
dismissAllNotifications -> dismissAll
getNotifications -> getHistory? or something similar

Also in the description below:

```
It should be given an ng-template element containing the elements to be displayed and an object containing properties to configure the notification.
```

It should be a `TemplateRef` they are giving to the function rather than the element. You should also make the options parameter optional, I notice in the example above you call:

```
notificationService.showNotification(myNotification, {})
```

Allow them to skip this second argument if they just want to use the default options.

donna-marijne commented:
Please assign your PRs to me and I will determine the reviewer.

donna-marijne commented on docs/app/pages/components/components-sections/notifications/notifications/notifications.component.html:
```diff
@@ -0,0 +1,61 @@
+
+<p>Notifications can be used to inform the user of occurring events or to alert the user of something that may require their
+    attention. Notifications will appear in the top right corner of the browser window and can be dismissed by the user by
+    clicking the close button.</p>
```
Position is configurable, so not necessarily top right. Also by default, notifications dismiss after 4 seconds (also configurable).

PearseMcMurray replied:
Documented that position and duration are configurable.

donna-marijne commented on docs/app/pages/components/components-sections/notifications/notifications/notifications.component.html:
```diff
@@ -0,0 +1,61 @@
+<uxd-api-properties>
+    <tr uxd-api-property name="position" type="string" defaultValue="topRight">
+        Sets the position of the list of notifications within the browser window.
+        The possible options are <code>topRight</code> and <code>topLeft</code>.
```
Include bottom left and bottom right if possible. Most apps notify bottom right currently.

PearseMcMurray replied:
Bottom left and right added.

PearseMcMurray commented on docs/app/pages/components/components-sections/notifications/notifications/notifications.component.html:
```diff
@@ -0,0 +1,61 @@
+    attention. Notifications will appear in the top right corner of the browser window and can be dismissed by the user by
+    clicking the close button.</p>
+
+<p>Notifications are controlled by the <code>notificationService</code> which can be injected into any controller in the application.</p>
```
Changed to "It is recommended that the service is injected into the AppComponent".

PearseMcMurray commented on docs/app/pages/components/components-sections/notifications/notifications/notifications.component.html:
```diff
@@ -0,0 +1,61 @@
+    <tr uxd-api-property name="direction" type="string" defaultValue="append">
+        Sets the order in which notifications are displayed. The possible options are <code>append</code> and <code>prepend</code>.
+        When set to <code>append</code> newer notifications will appear below older ones.
+        When set to <code>prepend</code> newer notifications will appear above older ones
```
Now using 'above' and 'below'.

PearseMcMurray commented on docs/app/pages/components/components-sections/notifications/notifications/notifications.component.html:
```diff
@@ -0,0 +1,61 @@
+        along with a <code>date</code> property which will contain the time at which the notification was shown.
+    </tr>
+    <tr uxd-api-property name="getNotifications">
+        This function will return an array of <code>NotificationRef</code> objects.
```
NotificationRef interface added to code snippets.

PearseMcMurray commented on docs/app/pages/components/components-sections/notifications/notifications/notifications.component.html:
```diff
@@ -0,0 +1,61 @@
+<uxd-api-properties>
+    <tr uxd-api-property name="showNotification">
+        This function should be called to show a notification.
+        It should be given an <code>ng-template</code> element containing the elements to be displayed and an object containing properties
```
Using bullet points where necessary.

PearseMcMurray commented on docs/app/pages/components/components-sections/notifications/notifications/notifications.component.html:
```diff
@@ -0,0 +1,61 @@
+    </tr>
+</uxd-api-properties>
+
+<p>The following options can be used to configure the location and behavior of the list of notifications:</p>
```
Rearranged text and tables for clarity.
Property table headers added.
Added explanation of use of ux-notifications-list.

PearseMcMurray commented:
Applied review comments to documentation.
Code snippets and Plunker example added.
AngularJS component marked as deprecatedand Hybrid tag removed. NotificationService removed from HybridModule.

ashley-hunter commented on docs/app/data/components-page.json:
```diff
@@ -709,10 +709,26 @@
             "title": "Notifications",
             "link": "notifications",
             "sections": [
+                {
```
Add dependency in here, they will need to add the brower animation module to their app module - would also be good to mention this to the documentation

ashley-hunter commented on docs/app/pages/components/components-sections/notifications/notifications/notifications.component.html:
```diff
@@ -0,0 +1,104 @@
+        along with a <code>date</code> property which will contain the time at which the notification was shown.
+    </tr>
+    <tr uxd-api-property name="getHistory">
+        This function will return an array of <code>NotificationRef</code> objects.
```
Possibly mention that this returns an array containing all the notifications that have been shown

ashley-hunter commented on docs/app/pages/components/components-sections/notifications/notifications/notifications.component.html:
```diff
@@ -0,0 +1,104 @@
+    The following functions can be called on the service:</p>
+
+<uxd-api-properties tableTitle="Notification Service">
+    <tr uxd-api-property name="show">
```
Consider using the args and returns properties on the uxd-api-property directive rather than describing the data types and arguments

ashley-hunter commented on docs/app/pages/components/components-sections/notifications/notifications/snippets/app.ts:
```diff
@@ -0,0 +1,24 @@
+
+}
+
+interface NotificationRef {
```
These should not be defined here, they should be imported from '@ux-aspects/ux-aspects'

ashley-hunter commented on docs/app/app.module.ts:
```diff
@@ -3,13 +3,15 @@ import * as angular from 'angular';
 import { TabsModule } from 'ngx-bootstrap/tabs';
 
 // Import UX Aspects
+// import { CheckboxModule, ColorServiceModule, EboxModule, FlippableCardModule, NotificationServiceModule, ProgressBarModule, RadioButtonModule, SparkModule, ToggleSwitchModule } from '../../src/index';
```
Remove this commented out code

ashley-hunter commented on docs/app/pages/components/components-sections/notifications/notifications/notifications.component.html:
```diff
@@ -0,0 +1,142 @@
+                        <ux-number-picker [(value)]="duration" min="0" max="3600000"></ux-number-picker>
+                    </div>
+                </div>
+                <div class="form-group">
```
This needs to be in a row and columns, eg:

```html
div class="row uxd-customize-row">
                    <div class="col-md-5 col-sm-12">
                        <label for="duration">Duration</label>
                        <ux-number-picker [(value)]="duration" min="0" max="3600000"></ux-number-picker>
                    </div>
                </div>

                <div class="row uxd-customize-row">
                    <div class="col-md-5 col-sm-12">
                        <label for="duration">Background Color</label>

                        <div class="input-group m-b-sm">
                              ...
                        </div>
                    </div>
                </div>

                <div class="row uxd-customize-row">
                    <div class="col-md-6 col-sm-12">
                        <p>
                            <small class="text-muted">Recommended Colors</small>
                        </p>
                        <div class="color-palette">
                            ...
                        </div>
                    </div>
                </div>
            </accordion-group>
```

ashley-hunter commented on docs/app/pages/components/components-sections/notifications/notifications/notifications.component.html:
```diff
@@ -0,0 +1,142 @@
+                <div class="form-group">
+                    <label class="form-label">Background Color</label>
+                    <div class="input-group m-b-sm">
+                        <input type="text" placeholder="Enter color" class="form-control" [(value)]="backgroundColor" text-string-color>
```
Remove `text-string-color` - this is an AngularJS directive with no effect here.

Also binding to `value` does not work here, you need to use `ngModel` instead.

ashley-hunter commented on docs/app/pages/components/components-sections/notifications/notifications/notifications.component.html:
```diff
@@ -0,0 +1,142 @@
+                    <label class="form-label">Background Color</label>
+                    <div class="input-group m-b-sm">
+                        <input type="text" placeholder="Enter color" class="form-control" [(value)]="backgroundColor" text-string-color>
+                        <span class="input-group-addon color-display" [ngStyle]="{ 'background-color': backgroundColor }">&nbsp;</span>
```
When binding to background color use:

```
[style.background-color]="backgroundColor"
```

There is also an nbsp inside here, remove this and set the size in the css instead

ashley-hunter commented on docs/app/pages/components/components-sections/notifications/notifications/notifications.component.html:
```diff
@@ -0,0 +1,142 @@
+                    </div>
+                    <small class="text-muted">Recommended Colors</small>
+                    <div class="color-palette">
+                        <div class="color-box" [ngStyle]="{ backgroundColor: notificationColors.primary }" (click)="setColor('primary')"></div>
```
Change the colors in the class to be an array:

```typescript
colors = [
        this.colorService.getColor('primary').toHex(),
        this.colorService.getColor('accent').toHex(),
        this.colorService.getColor('chart4').toHex(),
        this.colorService.getColor('chart5').toHex(),
        this.colorService.getColor('ok').toHex(),
        this.colorService.getColor('warning').toHex(),
        this.colorService.getColor('critical').toHex()
    ];
```

Then in our view we can avoid repetition:

```html
<div class="color-palette">
   <div class="color-box" *ngFor="let color of colors" [style.background-color]="color" (click)="backgroundColor = color"></div>
</div>
```

ashley-hunter commented on docs/app/pages/components/components-sections/notifications/notifications/notifications.component.html:
```diff
@@ -0,0 +1,142 @@
+                    <div class="color-palette">
+                        <div class="color-box" [ngStyle]="{ backgroundColor: notificationColors.primary }" (click)="setColor('primary')"></div>
+                        <div class="color-box" [ngStyle]="{ backgroundColor: notificationColors.accent }" (click)="setColor('accent')"></div>
+                        <div class="color-box" [ngStyle]="{ backgroundColor: notificationColors.chart4 }" (click)="setColor('chart4')"></div>
```
The reason none of these are showing is because there are no styles for them, eg:

```css
.color-palette {
    display: flex;
}

.color-box {
    width: 25px;
    height: 25px;
    margin: 0 3px;
    border: 2px solid rgba(0, 0, 0, 0.3);
    cursor: pointer;
}

.color-display {
    border: 2px solid rgba(0, 0, 0, 0.3);
    transition: background-color 0.3s linear;
    width: 34px;
}
```

ashley-hunter commented on docs/app/pages/components/components-sections/notifications/notifications/notifications.component.html:
```diff
@@ -0,0 +1,142 @@
+</blockquote>
+
+<blockquote>
+    <p><strong>Note</strong>: To display web animations in IE 11 the <code>web-animations-js</code> package must be imported.</p>
```
Web animations are also not available in Edge

ashley-hunter commented on docs/app/pages/components/components-sections/notifications/notifications/snippets/app.ts:
```diff
@@ -0,0 +1,30 @@
+        critical: this.colorService.getColor('critical').toHex()
+    };
+    
+    setColor (color: string): void {
```
This can be removed and done in the view

PearseMcMurray commented:
Applied code review comments.
Rearranged contents of the Customize Example panel.
Added styling for recommended colours.
List of recommended colours now in array.
Changed comment regarding importatation of web-animations-js package to mention Edge.
Added 'Angular/animations' to Plunker template.

ashley-hunter commented on docs/app/pages/components/components-sections/notifications/notifications/notifications.component.ts:
```diff
@@ -0,0 +1,55 @@
+            'app.component.ts': this.snippets.raw.appTs,
+            'app.component.css': this.snippets.raw.appCss,
+        },
+        modules: [{
```
Need to add import for BrowserAnimationsModule here

ashley-hunter commented on docs/app/services/plunker/templates/config_js.txt:
```diff
@@ -28,6 +28,7 @@ System.config({
     '@angular/platform-browser': 'https://unpkg.com/@angular/platform-browser/bundles/platform-browser.umd.js',
     '@angular/platform-browser-dynamic': 'https://unpkg.com/@angular/platform-browser-dynamic/bundles/platform-browser-dynamic.umd.js',
     '@angular/upgrade/static': 'https://unpkg.com/@angular/upgrade/bundles/upgrade-static.umd.js',
+    '@angular/animations': 'https://unpkg.com/@angular/animations/bundles/animations.umd.js',
```
Need to add `@angular/animations/browser` and `@angular/platform-browser/animations` packages in here also. That will fix the delay in animations

PearseMcMurray commented:
Changed duration unit to seconds.
Added animation packages to Plunker template.
